files/default/neutron-ha-tool.py

```diff
@@ -585,8 +589,54 @@ def migrate_router(qclient, router_id, agent, target):
     if router_id not in list_routers_on_l3_agent(qclient, target['id']):
         raise RuntimeError("Failed to add router_id=%s from agent_id=%s" %
                            (router_id, agent['id']))
+    if wait_for_ports:
+        wait_router_ports_active(qclient, router_id, target['host'])
 
 
+def wait_router_ports_active(qclient, router_id, target_host, maxtries=60):
+    """
+    Returns nothing. Waits for all non-distributed ports of a router
+    for being in the ACTIVE status on a specific host
+
+    :param qclient: A neutron client
+    :param router_id: The id of the router to check
+    :param target_host: The host on which the ports should be ACTIVE
+                        (reflected by the binding:host_id attribute of
+                        a port)
+    :param maxtries: Maximum number of times the ports' status is polled,
+                     an expection is raised there are still ports with a
+                     non ACTIVE status after that.
+    """
+
+    # FIXME: Don't check the "network:router_interface_distributed" ports
+    # as those are not hosted on the network node
+    router_port_list = qclient.list_ports(device_id=router_id, fields='id')
+    port_ids = [r['id'] for r in router_port_list['ports']]
+
+    LOG.info("Wait for the ports of router_id=%s to be ACTIVE" % router_id)
+    while (maxtries > 0) and (len(port_ids) > 0):
+        remaining = []
+        for port_id in port_ids:
+            port = qclient.show_port(port_id)['port']
+            LOG.debug("Router port=%s on hostid=%s in status=%s" %
+                      (port['id'], port['binding:host_id'],
+                       port['status']))
+            if ((port['binding:host_id'] == target_host) and
+                (port['status'] == 'ACTIVE')):
+                LOG.info("port_id=%s is ACTIVE on host_id=%s" %
+                         (port_id, target_host))
+            else:
+                remaining.append(port_id)
+
+        if len(remaining) > 0:
+            time.sleep(1)
+            maxtries-=1
+        port_ids = remaining
+
+    if len(port_ids):
+        raise RuntimeError("Some ports are not ACTIVE on router_id=%s, ports=[%s]" %
```

Comment:
I don't think you need to repeat the word `ports` here:

    raise RuntimeError("Some ports are not ACTIVE on router_id=%s: [%s]" %

Small nitpick but it's a pain when log lines are long enough to scroll off the right-hand edge of the terminal ...